client/src/Tools/_framework/ToolPanels/MoveContentToFolder.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect, useState } from "react";
+import {
+  Text,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  SimpleGrid,
+  Heading,
+  IconButton,
+  HStack,
+} from "@chakra-ui/react";
+import axios from "axios";
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { useFetcher } from "react-router-dom";
+import { ActivityStructure } from "../Paths/ActivityEditor";
+
+interface activeView {
+  folder: null | {
+    name: string;
+    id: number;
+    parentFolderId: number | null;
+  };
+  contents: {
+    isFolder: boolean;
+    name: string;
+    id: number;
+  }[];
+}
+
+export default function MoveContentToFolder({
+  isOpen,
+  onClose,
+  id,
+  currentParentId,
+}) {
+  // Set when the modal opens
+  const [parentId, setParentId] = useState<number | null>(null);
+  const [contentName, setContentName] = useState<string>("");
+  useEffect(() => {
+    if (isOpen) {
+      // TODO: does this async function needed to be awaited for before we end this effect?
```

Comment:
`useEffect()` cannot be asynchronous. It has to return either nothing or a function that is run at the end (to clean up side effects). There is nothing that is waiting for the `useEffect` function to finish. If you had code later in the `useEffect` function that you want to wait until `updateActiveView` is finished, then you'd have to structure it differently (e.g., write a separate asynchronous function/promise that is called from `useEffect`).

You can delete this `TODO`.